In [1]:
# pip install xgboost

  Obtaining dependency information for xgboost from https://files.pythonhosted.org/packages/24/ec/ad387100fa3cc2b9b81af0829b5ecfe75ec5bb19dd7c19d4fea06fb81802/xgboost-2.0.3-py3-none-win_amd64.whl.metadata
   ---------------------------------------- 0.0/99.8 MB ? eta -:--:--
   ---------------------------------------- 0.0/99.8 MB ? eta -:--:--
   ---------------------------------------- 0.0/99.8 MB 435.7 kB/s eta 0:03:49
   ---------------------------------------- 0.1/99.8 MB 871.5 kB/s eta 0:01:55
   ---------------------------------------- 0.2/99.8 MB 1.6 MB/s eta 0:01:02
   ---------------------------------------- 0.4/99.8 MB 2.0 MB/s eta 0:00:50
   ---------------------------------------- 0.5/99.8 MB 2.1 MB/s eta 0:00:47
   ---------------------------------------- 0.7/99.8 MB 2.3 MB/s eta 0:00:43
   ---------------------------------------- 0.8/99.8 MB 2.4 MB/s eta 0:00:41
   ---------------------------------------- 0.9/99.8 MB 2.5 MB/s eta 0:00:40
   --------------------------------

In [1]:
import numpy as np
import pandas as pd

from sklearn.model_selection import cross_val_score, GridSearchCV, KFold, RandomizedSearchCV, train_test_split

import xgboost as xgb

from sklearn.preprocessing import RobustScaler 

from sklearn.metrics import mean_squared_error

# 🏆 9 diamonds 
##  RSME = 547 
## competi

In [2]:
diamonds_1 = pd.read_csv('./data/diamonds_1.csv')
diamonds_2 = pd.read_csv('./data/diamonds_2.csv')
diamonds_3 = pd.read_csv('./data/diamonds_3.csv')
diamonds_4 = pd.read_csv('./data/diamonds_4.csv')
diamonds_5 = pd.read_csv('./data/diamonds_5.csv')
diamonds_6 = pd.read_csv('./data/diamonds_6.csv')
diamonds_7 = pd.read_csv('./data/diamonds_7.csv')
diamonds_8 = pd.read_csv('./data/diamonds_8.csv')
diamonds_9 = pd.read_csv('./data/diamonds_9.csv')

In [3]:
print(diamonds_1.shape)
print(diamonds_2.shape)
print(diamonds_3.shape)
print(diamonds_4.shape)
print(diamonds_5.shape)
print(diamonds_6.shape)
print(diamonds_7.shape)
print(diamonds_8.shape)
print(diamonds_9.shape)

(40439, 26)
(40439, 9)
(40455, 9)
(40439, 7)
(40439, 21)
(40439, 19)
(40439, 10)
(40439, 36)
(40455, 10)


In [4]:
X_1 = diamonds_1.drop('price', axis=1)
y_1 = diamonds_1['price']

X_2 = diamonds_2.drop('price', axis=1)
y_2 = diamonds_2['price']

X_3 = diamonds_3.drop('price', axis=1)
y_3 = diamonds_3['price']

X_4 = diamonds_4.drop('price', axis=1)
y_4 = diamonds_4['price']

X_5 = diamonds_5.drop('price', axis=1)
y_5 = diamonds_5['price']

X_6 = diamonds_6.drop('price', axis=1)
y_6 = diamonds_6['price']

X_7 = diamonds_7.drop('price', axis=1)
y_7 = diamonds_7['price']

X_8 = diamonds_8.drop('price', axis=1)
y_8 = diamonds_8['price']

X_9 = diamonds_9.drop('price', axis=1)
y_9 = diamonds_9['price']

# ⚠️ NO SCALER ⚠️

## Regression

In [5]:
X_1_train, X_1_test, y_1_train, y_1_test = train_test_split(X_1, y_1, test_size=0.2, random_state=42)
X_2_train, X_2_test, y_2_train, y_2_test = train_test_split(X_2, y_2, test_size=0.2, random_state=42)
X_3_train, X_3_test, y_3_train, y_3_test = train_test_split(X_3, y_3, test_size=0.2, random_state=42)
X_4_train, X_4_test, y_4_train, y_4_test = train_test_split(X_4, y_4, test_size=0.2, random_state=42)
X_5_train, X_5_test, y_5_train, y_5_test = train_test_split(X_5, y_5, test_size=0.2, random_state=42)
X_6_train, X_6_test, y_6_train, y_6_test = train_test_split(X_6, y_6, test_size=0.2, random_state=42)
X_7_train, X_7_test, y_7_train, y_7_test = train_test_split(X_7, y_7, test_size=0.2, random_state=42)
X_8_train, X_8_test, y_8_train, y_8_test = train_test_split(X_8, y_8, test_size=0.2, random_state=42)
X_9_train, X_9_test, y_9_train, y_9_test = train_test_split(X_9, y_9, test_size=0.2, random_state=42)

In [15]:
model = xgb.XGBRegressor(objective="reg:squarederror", random_state=42)

In [16]:
# model.fit(X_1_train, y_1_train)
# model.fit(X_2_train, y_2_train) 
# model.fit(X_3_train, y_3_train)
# model.fit(X_4_train, y_4_train)
# model.fit(X_5_train, y_5_train)
# model.fit(X_6_train, y_6_train)
model.fit(X_7_train, y_7_train)
# model.fit(X_8_train, y_8_train)
# model.fit(X_9_train, y_9_train)

XGBRegressor(base_score=None, booster=None, callbacks=None,
             colsample_bylevel=None, colsample_bynode=None,
             colsample_bytree=None, device=None, early_stopping_rounds=None,
             enable_categorical=False, eval_metric=None, feature_types=None,
             gamma=None, grow_policy=None, importance_type=None,
             interaction_constraints=None, learning_rate=None, max_bin=None,
             max_cat_threshold=None, max_cat_to_onehot=None,
             max_delta_step=None, max_depth=None, max_leaves=None,
             min_child_weight=None, missing=nan, monotone_constraints=None,
             multi_strategy=None, n_estimators=None, n_jobs=None,
             num_parallel_tree=None, random_state=42, ...)

In [17]:
# pred_1 = model.predict(X_1_test)
# pred_2 = model.predict(X_2_test)
# pred_3 = model.predict(X_3_test)
# pred_4 = model.predict(X_4_test)
# pred_5 = model.predict(X_5_test)
# pred_6 = model.predict(X_6_test)
pred_7 = model.predict(X_7_test)
# pred_8 = model.predict(X_8_test)
# pred_9 = model.predict(X_9_test)

In [18]:
# rmse_1 = mean_squared_error(y_1_test, pred_1, squared=False)
# rmse_2 = mean_squared_error(y_2_test, pred_2, squared=False)
# rmse_3 = mean_squared_error(y_3_test, pred_3, squared=False)
# rmse_4 = mean_squared_error(y_4_test, pred_4, squared=False)
# rmse_5 = mean_squared_error(y_5_test, pred_5, squared=False)
# rmse_6 = mean_squared_error(y_6_test, pred_6, squared=False)
rmse_7 = mean_squared_error(y_7_test, pred_7, squared=False)
# rmse_8 = mean_squared_error(y_8_test, pred_8, squared=False)
#rmse_9 = mean_squared_error(y_9_test, pred_9, squared=False)

C:\Users\tete1\miniconda3\envs\jupyter_env\lib\site-packages\sklearn\metrics\_regression.py:483: FutureWarning: 'squared' is deprecated in version 1.4 and will be removed in 1.6. To calculate the root mean squared error, use the function'root_mean_squared_error'.
  warnings.warn(


In [19]:
print(rmse_7)
print(rmse_8)
print(rmse_9)

550.8063892222714
596.8209675239349
547.8617028978986


In [103]:
print(rmse_1)
print(rmse_2)
print(rmse_3)
print(rmse_4)
print(rmse_5)
print(rmse_6)
print(rmse_7)

584.7097780680875
544.3640587464998
597.8630398338951
556.2992317711048
580.0749165543122
564.4505993988597
550.8063892222714


## Winner: 9 diamonds - RSME 547

### Hyperparameters search

In [ ]:
# Info https://www.kaggle.com/code/stuarthallows/using-xgboost-with-scikit-learn

# ⚠️ ROBUST SCALER ⚠️

In [88]:
robust_scaler = RobustScaler()

In [89]:
# X_1_rob = robust_scaler.fit_transform(X_1)
# X_2_rob = robust_scaler.fit_transform(X_2)
# X_3_rob = robust_scaler.fit_transform(X_3)
# X_4_rob = robust_scaler.fit_transform(X_4)
# X_5_rob = robust_scaler.fit_transform(X_5)
# X_6_rob = robust_scaler.fit_transform(X_6)
X_7_rob = robust_scaler.fit_transform(X_7)

In [90]:
# X_1_rob_train, X_1_rob_test, y_1_rob_train, y_1_rob_test = train_test_split(X_1_rob, y_1, test_size=0.2, random_state=42)
# X_2_rob_train, X_2_rob_test, y_2_rob_train, y_2_rob_test = train_test_split(X_2_rob, y_2, test_size=0.2, random_state=42)
# X_3_rob_train, X_3_rob_test, y_3_rob_train, y_3_rob_test = train_test_split(X_3_rob, y_3, test_size=0.2, random_state=42)
# X_4_rob_train, X_4_rob_test, y_4_rob_train, y_4_rob_test = train_test_split(X_4_rob, y_4, test_size=0.2, random_state=42)
# X_5_rob_train, X_5_rob_test, y_5_rob_train, y_5_rob_test = train_test_split(X_5_rob, y_5, test_size=0.2, random_state=42)
# X_6_rob_train, X_6_rob_test, y_6_rob_train, y_6_rob_test = train_test_split(X_6_rob, y_6, test_size=0.2, random_state=42)
X_7_rob_train, X_7_rob_test, y_7_rob_train, y_7_rob_test = train_test_split(X_7_rob, y_7, test_size=0.2, random_state=42)

In [91]:
model = xgb.XGBRegressor(objective="reg:squarederror", random_state=42)

In [92]:
# model.fit(X_1_rob_train, y_1_rob_train)
# model.fit(X_2_rob_train, y_2_rob_train)
# model.fit(X_3_rob_train, y_3_rob_train)
# model.fit(X_4_rob_train, y_4_rob_train)
# model.fit(X_5_rob_train, y_5_rob_train)
# model.fit(X_6_rob_train, y_6_rob_train)
model.fit(X_7_rob_train, y_7_rob_train)

XGBRegressor(base_score=None, booster=None, callbacks=None,
             colsample_bylevel=None, colsample_bynode=None,
             colsample_bytree=None, device=None, early_stopping_rounds=None,
             enable_categorical=False, eval_metric=None, feature_types=None,
             gamma=None, grow_policy=None, importance_type=None,
             interaction_constraints=None, learning_rate=None, max_bin=None,
             max_cat_threshold=None, max_cat_to_onehot=None,
             max_delta_step=None, max_depth=None, max_leaves=None,
             min_child_weight=None, missing=nan, monotone_constraints=None,
             multi_strategy=None, n_estimators=None, n_jobs=None,
             num_parallel_tree=None, random_state=42, ...)

In [93]:
# pred_1_rob = model.predict(X_1_rob_test)
# pred_2_rob = model.predict(X_2_rob_test)
# pred_3_rob = model.predict(X_3_rob_test)
# pred_4_rob = model.predict(X_4_rob_test)
# pred_5_rob = model.predict(X_5_rob_test)
# pred_6_rob = model.predict(X_6_rob_test)
pred_7_rob = model.predict(X_7_rob_test)

In [94]:
# rmse_1_rob = mean_squared_error(y_1_rob_test, pred_1_rob, squared=False)
# rmse_2_rob = mean_squared_error(y_2_rob_test, pred_2_rob, squared=False)
# rmse_3_rob = mean_squared_error(y_3_rob_test, pred_3_rob, squared=False)
# rmse_4_rob = mean_squared_error(y_4_rob_test, pred_4_rob, squared=False)
# rmse_5_rob = mean_squared_error(y_5_rob_test, pred_5_rob, squared=False)
# rmse_6_rob = mean_squared_error(y_6_rob_test, pred_6_rob, squared=False)
rmse_7_rob = mean_squared_error(y_7_rob_test, pred_7_rob, squared=False)

C:\Users\tete1\miniconda3\envs\jupyter_env\lib\site-packages\sklearn\metrics\_regression.py:483: FutureWarning: 'squared' is deprecated in version 1.4 and will be removed in 1.6. To calculate the root mean squared error, use the function'root_mean_squared_error'.
  warnings.warn(


In [96]:
print(rmse_1_rob)
print(rmse_2_rob)
print(rmse_3_rob)
print(rmse_4_rob)
print(rmse_5_rob)
print(rmse_6_rob)
print(rmse_7_rob)

584.7097780680875
544.3640587464998
597.8630398338951
556.2992317711048
580.0749165543122
564.4505993988597
550.8063892222714


## 🏅 Winner: 2do diamonds RSME = 544.36

# Now: Diamonds_test + NO Scaler

In [97]:
diamonds_test = pd.read_csv('./data/diamonds_test.csv')

In [104]:
diamonds_test.drop('depth', axis=1, inplace=True)

In [105]:
diamonds_test.drop('city', axis=1, inplace=True)

In [106]:
cut_map = {'Fair': 1,
           'Good': 2,
           'Very Good': 3,
           'Premium': 4,
           'Ideal': 5}

In [107]:
diamonds_test['cut'] = diamonds_test['cut'].map(cut_map)

In [108]:
color_map = {'J': 1,
             'I': 2,
             'H': 3,
             'G': 4,
             'F': 5,
             'E': 6,
             'D': 7}

In [109]:
diamonds_test['color'] = diamonds_test['color'].map(color_map)

In [110]:
clarity_map = {'I1': 1,
               'SI2': 2,
               'SI1': 3,
               'VS2': 4,
               'VS1': 5,
               'VVS2': 6,
               'VVS1': 7,
               'IF': 8}

In [111]:
diamonds_test['clarity'] = diamonds_test['clarity'].map(clarity_map)

In [112]:
diamonds_test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13485 entries, 0 to 13484
Data columns (total 9 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   id       13485 non-null  int64  
 1   carat    13485 non-null  float64
 2   cut      13485 non-null  int64  
 3   color    13485 non-null  int64  
 4   clarity  13485 non-null  int64  
 5   table    13485 non-null  float64
 6   x        13485 non-null  float64
 7   y        13485 non-null  float64
 8   z        13485 non-null  float64
dtypes: float64(5), int64(4)
memory usage: 948.3 KB


In [113]:
diamonds_test.drop('id', axis=1, inplace=True)

In [114]:
diamonds_2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40439 entries, 0 to 40438
Data columns (total 9 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   price    40439 non-null  int64  
 1   carat    40439 non-null  float64
 2   table    40439 non-null  float64
 3   x        40439 non-null  float64
 4   y        40439 non-null  float64
 5   z        40439 non-null  float64
 6   cut      40439 non-null  int64  
 7   color    40439 non-null  int64  
 8   clarity  40439 non-null  int64  
dtypes: float64(5), int64(4)
memory usage: 2.8 MB


In [115]:
diamonds_test = diamonds_test[['carat', 'table', 'x', 'y', 'z', 'cut', 'color', 'clarity']]

In [116]:
pred_test = model.predict(diamonds_test)

In [117]:
pred_test = pd.DataFrame(pred_test, columns=['price'])

In [118]:
pred_test['id'] = range(0, len(pred_test))

In [119]:
pred_test = pred_test[['id', 'price']]

In [120]:
pred_test.to_csv('./submissions/sub_XGBRegressor_544.csv', index=False)